test/any/web_catalog/api-extractor-integration.es6.js

```diff
@@ -0,0 +1,395 @@
+        expect(apiCatalog.Window.sort()).toEqual(['Function', 'Object',
+          'FunctionInterface', 'DuplicateFunctionInterface', 'ObjectInterface',
+          'ObjectLibrary', 'nonObjectLibrary', 'FunctionNonInterface',
+          'constant'].sort());
+      });
+    it('extract api for Object and Function without filter build-in apis.',
+      function() {
+        expect(apiCatalog.Object.sort()).toEqual(['arguments',
+          'constructor', 'toLocaleString', 'toString', 'valueOf'].sort());
+        expect(apiCatalog.Function.sort()).toEqual([
+          'prototype', 'caller', 'length', 'name'].sort());
+      });
+    it('only includes Function/Object with meaningful properties.', function() {
+      expect(apiCatalog.FunctionNonInterface).toBeUndefined();
+      expect(apiCatalog.FunctionInterface).toBeDefined();
+      expect(apiCatalog.nonObjectLibrary).toBeUndefined();
+      expect(apiCatalog.ObjectLibrary).toBeDefined();
+      expect(apiCatalog.ObjectInterface).toBeDefined();
+    });
+    it('filter build-in apis for non-special cases.', function() {
+      expect(apiCatalog.FunctionNonInterface).not.toContain('caller');
+      expect(apiCatalog.FunctionNonInterface).not.toContain('length');
+      expect(apiCatalog.FunctionNonInterface).not.toContain('name');
+      expect(apiCatalog.ObjectInterface).not.toContain('constructor');
+      expect(apiCatalog.ObjectInterface).not.toContain('toString');
+      expect(apiCatalog.ObjectInterface).not.toContain('toLocaleString');
+      expect(apiCatalog.ObjectInterface).not.toContain('valueOf');
+      expect(apiCatalog.ObjectLibrary).not.toContain('constructor');
+      expect(apiCatalog.ObjectLibrary).not.toContain('toString');
+      expect(apiCatalog.ObjectLibrary).not.toContain('toLocaleString');
+      expect(apiCatalog.ObjectLibrary).not.toContain('valueOf');
+    });
+    it('extract interfeaces even they are same object.', function() {
+      expect(apiCatalog.FunctionInterface).toBeDefined();
+      expect(apiCatalog.DuplicateFunctionInterface).toBeDefined();
+        expect(apiCatalog.DuplicateFunctionInterface).toEqual(
+          apiCatalog.FunctionInterface);
+    });
+    it('extract __proto__ as interface if it includes meaningful apis',
+      function() {
+        expect(apiCatalog.ProtoInterface).toBeDefined();
+        expect(apiCatalog.ProtoInterface).toContain('meaningfulAPI');
+      }
+    );
+    it('add instances\' api to its class if class is not Object', function() {
+      expect(apiCatalog.HiddenInterface).toBeDefined();
+      expect(apiCatalog.HiddenInterface).toEqual(['hiddenAPI']);
+      // If instances class is Object, do not add it to Object interface.
+      expect(apiCatalog.Object).not.toContain('hiddenAPI');
+    });
+    it('only includes own properties.', function() {
+      expect(apiCatalog.HiddenInterface).not.toContain('notwnProperty');
+    });
+    it('const primitive are filtered, constant object are not.', function() {
```

Comment:
nit: `filters out const primitives, but not const objects` to read "It filters out const primitives, but not const objects".